memory-viz/src/typeguards.ts

```diff
@@ -3,7 +3,7 @@ import { Style } from "./types";
 export function isArrayOfType<T>(value: any, type: string): value is T[] {
```

Comment:
So leaving this change as-is will conflict with the changes I've asked @MaxChen102 to make (about strict null checking).

Instead, I think the approach suggested in the last meeting of having two separate functions is best. Leave this one unchanged, and add a new `isArrayOfNullableType` function. Note that the resulting type from the new typeguard function should be `(T | null)[]` instead of just `T[]`.